### PR TITLE
Multipart returning 400

### PR DESCRIPTION
**Pull request by xsawyerx**
This is just a test + cleanups at first.

I'm going to now add the fix for it.

### CODE REVIEW

File: t/multipart_content.t [hunk 1]
```diff
@@ -15,39 +20,42 @@ use Dancer2;
...
-        my $res = $cb->(HTTP::Request->new($http, '/', $headers));
-        is($res->code, 200);
-    }
-    { # passes
-        my $headers = [ 'Content-Type' => 'text/html; charset=UTF-8' ];
-        my $res = $cb->(HTTP::Request->new($http, '/', $headers));
-        is($res->code, 200);
-    }
+    is(
+        $cb->( GET '/' => ['Content-Type' => 'multipart/form-data'] )->code,
```
Comment (veryrusty) [inline t/multipart_content.t]: The syntax for the extra headers isn't correct.  Should be:
```
$cb->( GET '/', 'Content-Type' => 'multipart/form-data' )->code,
```

**Comment by veryrusty** [comment]
Also look at #1509 (These tests with correct header syntax pass on that branch).

**Comment by cromedome** [comment]
:+1: for merging this test. Thanks @xsawyerx.

**Comment by veryrusty** [comment]
Trying to sort out the test failure for the April 2020 release.

If I grok this correctly; the 3rd test (https://github.com/PerlDancer/Dancer2/pull/1510/files#diff-1044d466c97c916209b2eab15abab50aR42) has a header that isn't what the test expects. The PSGI env gets
```
CONTENT_TYPE => 'Content-Type: multipart/form-data boundary=------boundary-------'
```
When the request is parsed; HTTP::Entity::Parser matches that against the registered types. None of those begin with `Content-Type`, so the request gets parsed as an OctetStream, and no error occurs.  ( Those details are here: https://github.com/kazeburo/HTTP-Entity-Parser/blob/master/lib/HTTP/Entity/Parser.pm#L53-L63 ).

The 4th test has a content type header that matches a registered parser. That it handed off to HTTP::MultiPartParser, which throws an error in parsing the request. With #1509 merged, the response is now a 400 (which I believe is what #1507 wanted). The error returned (generated by HTTP::MultiPartParse) is `End of stream encountered while parsing preamble`. 

To clean this up:
Delete the current 3rd test; it isn't doing what we expect.  The current 4th test should check it gets a 400 response.

**Comment by cromedome** [comment]
Resolved by #1509.

**Pull request closed without merging**